http/client.go

```diff
@@ -0,0 +1,68 @@
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// NewWithCert creates a new http.Client with a custom certificate
+func NewWithCert(CABundleFile, CABundleDir string) (*http.Client, error) {
+	// go default http transport settings
+	transport := &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		DialContext:           (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	if CABundleFile != "" || CABundleDir != "" {
+		certs, err := getCertPool(CABundleFile, CABundleDir)
+		if err != nil {
+			return nil, err
+		}
+		transport.TLSClientConfig = &tls.Config{RootCAs: certs}
+	}
+
+	return &http.Client{
+		Transport: transport,
```

Comment:
It would be nice having a configurable timeout.

Reply:
Yes, although I try to keep this PR only about the certificates.
We can do it in another PR.